include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_reduce_then_scan.h

```diff
@@ -724,6 +715,34 @@ struct __parallel_reduce_then_scan_scan_submitter<
     _InitType __init;
 };
 
+// We accept a set of variadic types to disambiguate between the different scan kernels. The set
+// of template parameters for __parallel_transform_reduce_then_scan here is expected to be used.
+template <typename _ExecutionPolicy, typename... ParamTypes>
+struct __reduce_then_scan_kernels
+{
+    using _CustomName = oneapi::dpl::__internal::__policy_kernel_name<_ExecutionPolicy>;
+    using _ReduceKernel = oneapi::dpl::__par_backend_hetero::__internal::__kernel_name_generator<
+        __reduce_then_scan_reduce_kernel, _CustomName, ParamTypes...>;
+    using _ScanKernel = oneapi::dpl::__par_backend_hetero::__internal::__kernel_name_generator<
+        __reduce_then_scan_scan_kernel, _CustomName, ParamTypes...>;
+    explicit __reduce_then_scan_kernels(const _ExecutionPolicy& __exec)
+        : __exec(__exec)
+        , __kernels(__internal::__kernel_compiler<_ReduceKernel, _ScanKernel>::__compile(__exec))
+    {
+    }
+    sycl::kernel __get_reduce_kernel() const { return __kernels[0]; }
+    sycl::kernel __get_scan_kernel() const { return __kernels[1]; }
+    bool __is_compiled_sg32() const
+    {
+        return oneapi::dpl::__internal::__kernel_sub_group_size(__exec, __get_reduce_kernel()) == std::uint32_t{32} &&
+               oneapi::dpl::__internal::__kernel_sub_group_size(__exec, __get_scan_kernel()) == std::uint32_t{32};
+    }
+private:
+    // idx 0 is the reduce kernel and idx 1 is the scan kernel
+    std::array<sycl::kernel, 2> __kernels;
```

Comment:
Is the array necessary?
Can we just construct a two public member fields:
`__reduce_kernel` and `__scan_kernel` then reference them directly?